### PR TITLE
Updating firebase/index.d.ts

### PR DESCRIPTION
**Pull request by schmidt-sebastian**
Updating the types in firebase/index.d.ts to pull in the recent API changes.

Fixes: https://github.com/firebase/firebase-js-sdk/issues/498

### CODE REVIEW

File: packages/firebase/index.d.ts [hunk 1]
```diff
@@ -617,15 +617,14 @@ declare namespace firebase.firestore {
 
   export type LogLevel = 'debug' | 'error' | 'silent';
 
-  function setLogLevel(logLevel: LogLevel): void;
+  export function setLogLevel(logLevel: LogLevel): void;
```
Comment (jshcrowthe) [inline packages/firebase/index.d.ts]: This is not needed as it is part of the `firebase` namespace which is exported at the end.

File: packages/firebase/index.d.ts [hunk 2]
```diff
@@ -698,9 +697,27 @@ declare namespace firebase.firestore {
     batch(): WriteBatch;
 
     /**
-     * The `firebase.app.App` associated with this `Firestore` instance.
+     * The `FirebaseApp` associated with this `Firestore` instance.
      */
-    app: firebase.app.App;
+    app: any;
```
Comment (jshcrowthe) [inline packages/firebase/index.d.ts]: Why did we remove the type?

Comment (schmidt-sebastian) [reply packages/firebase/index.d.ts]: Sorry, that was unintended.

File: packages/firebase/index.d.ts (hunk 1 above)
Comment (schmidt-sebastian) [reply packages/firebase/index.d.ts]: Before I change this, do you know if this is causing a problem?

I would really like if these typings matched the ones in the firestore.d.ts file and having this extra 'export' seems like a step in that direction.

Comment (jshcrowthe) [reply packages/firebase/index.d.ts]: This is fine. I looked and it seems to have no impact.

**Pull request merged**